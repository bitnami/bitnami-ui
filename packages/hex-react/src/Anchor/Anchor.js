import React from 'react';
import PropTypes from 'prop-types';
import cs from 'classnames';

const Anchor = props => {
  const {
    external,
    noReferrer,
    inverse,
    secondary,
    noUnderline,
    className,
    children,
    href,
    ...attributes
  } = props;

  const rel = cs({
    noopener: external === true,
    noreferrer: noReferrer === true,
  });

  const css = cs(className, {
    inverse: inverse === true,
    secondary: secondary === true,
    'no-decoration': noUnderline === true,
  });

  return React.createElement(
    'a',
    {
      href,
      className: css !== '' ? css : null,
      rel: rel !== '' ? rel : null,
      target: external === true ? '_blank' : null,
      ...attributes,
    },
    children,
  );
};

Anchor.propTypes = {
  /**
   * URL or hash for the anchor
   */
  href: PropTypes.string,
  /**
   * Open the anchor in a new tab. It adds the "noopener" protection
   */
  external: PropTypes.bool,
  /**
   * Disable the referrer header during the navigation
   */
  noReferrer: PropTypes.bool,
  /**
   * Use a light color. Ideal for dark backgrounds
   */
  inverse: PropTypes.bool,
  /**
   * Use the brand color for the anchor
   */
  secondary: PropTypes.bool,
  /**
   * Remove the underline for the anchor
   */
  noUnderline: PropTypes.bool,
  /**
   * Other CSS classes for the component
   */
  className: PropTypes.string,
  /**
   * Any element that will be wrapped by the Box
   */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
};

Anchor.defaultProps = {
  href: '',
  external: false,
  noReferrer: false,
  inverse: false,
  secondary: false,
  noUnderline: false,
  className: '',
  children: undefined,
};

export default Anchor;
