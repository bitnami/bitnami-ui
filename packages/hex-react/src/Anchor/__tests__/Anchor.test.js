import React from 'react';
import { shallow } from 'enzyme';

import Anchor from '../Anchor';

describe('Anchor', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Anchor href="https://design.bitnami.com">Test</Anchor>);
    expect(wrapper).toMatchSnapshot();
  });

  it('shows the children', () => {
    const text = 'Design';
    const wrapper = shallow(<Anchor href="https://design.bitnami.com">{text}</Anchor>);
    expect(wrapper.text()).toEqual(text);
  });

  it('adds no-decoration CSS class', () => {
    const wrapper = shallow(
      <Anchor href="https://design.bitnami.com" noUnderline>
        Test
      </Anchor>,
    );
    expect(wrapper.hasClass('no-decoration')).toBe(true);
  });

  it('adds secondary CSS class', () => {
    const wrapper = shallow(
      <Anchor href="https://design.bitnami.com" secondary>
        Test
      </Anchor>,
    );
    expect(wrapper.hasClass('secondary')).toBe(true);
  });

  it('adds inverse CSS class', () => {
    const wrapper = shallow(
      <Anchor href="https://design.bitnami.com" inverse>
        Test
      </Anchor>,
    );
    expect(wrapper.hasClass('inverse')).toBe(true);
  });

  it('sets the proper rel attribute for external Anchors', () => {
    const wrapper = shallow(
      <Anchor href="https://design.bitnami.com" external>
        Test
      </Anchor>,
    );
    expect(wrapper.prop('rel')).toEqual('noopener');
  });

  it('sets the proper target attribute for external Anchors', () => {
    const wrapper = shallow(
      <Anchor href="https://design.bitnami.com" external>
        Test
      </Anchor>,
    );
    expect(wrapper.prop('target')).toEqual('_blank');
  });

  it('sets the proper rel attribute for noreferrer', () => {
    const wrapper = shallow(
      <Anchor href="https://design.bitnami.com" noReferrer>
        Test
      </Anchor>,
    );
    expect(wrapper.prop('rel')).toEqual('noreferrer');
  });

  it('sets the proper rel attribute for external and noreferrer Anchors', () => {
    const wrapper = shallow(
      <Anchor href="https://design.bitnami.com" external noReferrer>
        Test
      </Anchor>,
    );
    expect(wrapper.prop('rel')).toEqual('noopener noreferrer');
  });
});
