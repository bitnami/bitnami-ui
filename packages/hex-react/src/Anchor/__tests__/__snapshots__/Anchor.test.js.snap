// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor renders correctly 1`] = `
<a
  className={null}
  href="https://design.bitnami.com"
  rel={null}
  target={null}
>
  Test
</a>
`;
